### PR TITLE
Array binary search

### PR DESCRIPTION
**Pull request by lucasjwilber**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401challenges/src/test/java/code401challenges/java/BinarySearchTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+public class BinarySearchTest {
+
+    @Test public void testBinarySearch() {
+        int[] testArray = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+        assertEquals(0, BinarySearch.binarySearchForValue(testArray, 1));
+
+        int[] testArray2 = {11,22,33,44,55,66,77};
+        assertEquals(-1, BinarySearch.binarySearchForValue(testArray, 999));
```
Comment (SharinaS) [inline code401challenges/src/test/java/code401challenges/java/BinarySearchTest.java]: Each assertEquals() belongs in its own `@Test public void test-cool-method-name() {}`. This will help you out a lot when tests start failing.

Comment (lucasjwilber) [reply code401challenges/src/test/java/code401challenges/java/BinarySearchTest.java]: Good advice. Split them up and gave them more descriptive names.
